### ISSUE TITLE
SP's with an Apostrophe in the name are displayed with &#39; 

### ISSUE THREAD

**Issue by trsau** [ausaccessfed/discovery-service#66]
I have a Simple attribute reflector in test called "TRS's Testing tool" which appears on the DS page as "Login to TRS&#39;s Testing tool" with a description of "TRS&#39;s Testing tool"

The mdui entry for this SP in the new metadata feed has the apostrophe.

I expect other special characters in SP's names and descriptions will have these characters transformed.

**Comment by bradleybeddoes** [comment]
I'll look into this @trsau (likely on/after Thu 31/3). 

From initial discussion with @smangelsdorf we believe it is likely within some escape code to prevent XSS in the DS app display layer as opposed to anything in the metadata/json layer (as you note that is represented correctly).

**Comment by bradleybeddoes** [comment]
Having dug through the code base and checked inputs / persistence etc I've determined that the problem lies at:

https://github.com/ausaccessfed/discovery-service/blob/develop/assets/javascripts/application.js#L164

The view being incorrect is simply due to HTML encoded strings being passed verbatim to a JSON decode routine.

The solution for this needs to be XSS safe given we can't trust what comes via Metadata feeds. I've done some research on this and come up with the following JS (http://stackoverflow.com/a/1395954 is what really set me on the right path):

```
function decodeEntityList(encodedString) {
    var textArea = document.createElement('textarea');
    textArea.innerHTML = encodedString;
    decodedString = textArea.value; 
    
    if ('remove' in Element.prototype) 
        textArea.remove();

    return decodedString;
}
```

I'll be addressing this in code but cc @rianniello / @smangelsdorf as FYI.

**Comment by bradleybeddoes** [comment]
Fixed by #67
